purchase_security/views/purchase_order_views.xml

```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="purchase_order_form" model="ir.ui.view">
+        <field name="name">purchase.order.form (in purchase_security)</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form" />
+        <field name="arch" type="xml">
+            <field name="user_id" position="before">
+                <field name="is_user_id_editable" invisible="1" />
+            </field>
+            <field name="user_id" position="attributes">
+                <attribute
+                    name="readonly"
+                >[('is_user_id_editable','=',False)]</attribute>
```

Comment:
For V17: `readonly, invisible, required` use Python expression instead of `attrs[(field, operator, value)]`
```suggestion
                >not is_user_id_editable</attribute>
```